File: solutions/csharp/etl/1/Etl.cs
using System;
using System.Collections.Generic;

public static class Etl
{
    public static Dictionary<string, int> Transform(Dictionary<int, string[]> old)
    {
        var letters = new Dictionary<string, int>();
        foreach (var score in old)
        {
            foreach (var letter in score.Value)
            {
                letters[letter.ToLower()] = score.Key;
            }
        }
        return letters;
    }
}

File: solutions/fsharp/etl/1/Etl.fs
module Etl

let transform (scoresWithLetters: Map<int, char list>): Map<char, int> = 
    scoresWithLetters |> Map.toSeq
    |> Seq.collect (fun (score, letters) -> letters |> Seq.map (fun letter -> System.Char.ToLower(letter), score))
    |> Map.ofSeq